packages/block-library/src/file/save.js

```diff
@@ -30,6 +32,7 @@ export default function save( { attributes } ) {
 						href={ href }
 						className="wp-block-file__button"
 						download={ true }
+						aria-describedby={ fileId }
```

Comment:
This is the main event! We're adding an `aria-describedby` prop to the Download button that references the file link above. That's how the Download button announces the file name from the link.